Fix some glibc FP routines for non-F

palmer-dabbelt opened the pull request:
Some glibc floating point routines use inline assembly.  This changes
them to use regular C code to do this when compiled without hardware
floatint-point support.  The change just uses regular C to perform
these operations when compiled on ABIs without hardware floating
point.

aswaterman commented:
The proposed fmin/fmax implementations return NaN when passed (NaN, non-NaN).  This isn't what IEEE 754-2008 minNum/maxNum want.  I guess I don't know if C11 wants fmin to behave like 754 minNum.  Mind checking that out?

palmer-dabbelt commented:
According to "man fmax", it's

````
RETURN VALUE
       These functions return the maximum of x and y.

       If one argument is a NaN, the other argument is returned.

       If both arguments are NaN, a NaN is returned.
````

I've fixed the relevant implementations (fdim returns NaN when there's a NaN, so it stays the same) and updated this PR.

aswaterman commented:
Thanks.
